### PR TITLE
feat(tasks): add task and attempt lineage skeleton

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_691fca79d75c4e6793f8d2ef` (agent: `codex`).

**External task ID**: awf-wave3-task-attempt-lineage-20260426T055128Z

### Task
Implement the next PRD lifecycle-model slice: introduce first-class task and task-attempt persistence while preserving the current workspace-backed execution model.

Problem:
- The PRD says task, task attempt, workspace, and operation should be distinct lifecycle objects.
- Current AWF treats a workspace as the task. That is good enough for execution, but bad for retry lineage, redispatch, supersession, and future scheduling.

Required behavior for this slice:
- Add `tasks` and `task_attempts` DB models plus Alembic migration.
- On v2 workspace creation, create or reuse a task row keyed by request external_id/idempotency where available, then create one task_attempt row linked to the workspace.
- Attempt should store workspace_id, attempt_number, agent, status/current workspace status snapshot, created_at/updated_at, and enough repo/base/title metadata to support `/v1/tasks` without losing compatibility.
- Keep existing workspace fields and API responses backward compatible.
- Update `/v1/tasks` to expose task_id and current/latest attempt from the new tables when present, falling back to workspace-backed rows for historical data.
- Do not rewrite scheduling or retries in this slice.

TDD requirements:
- Add failing DB/service/API tests first: v2 create creates task + attempt, repeated idempotent create does not duplicate attempts, task listing works for new rows and legacy workspace-only rows, attempt numbers increment for a manually-created second attempt fixture.

Validation commands:
- uv run ruff check src/awf/db src/awf/api/routes/tasks.py src/awf/api/schemas.py src/awf/service/workspaces.py migrations tests/unit/db tests/unit/api/test_tasks.py tests/unit/service/test_task_attempts.py
- uv run mypy src/awf/db src/awf/api/routes/tasks.py src/awf/api/schemas.py src/awf/service/workspaces.py
- uv run pytest tests/unit/db/test_task_attempts.py tests/unit/api/test_tasks.py tests/unit/service/test_task_attempts.py -q

Strict TDD and AWF contract:
- Write focused failing tests first and commit the red tests, including the failure summary in the commit body.
- Implement until green, then run the validation commands listed below.
- Commit locally with conventional commits.
- Do not push, switch branches, rebase, open PRs, merge PRs, or resolve GitHub threads manually outside the task contract. AWF owns those lifecycle steps.
- Keep the change tightly scoped to the owned paths.

---
Validation: 6 profile command(s) passed inside the workspace container.

### CODE REVIEW

File: src/awf/db/repositories.py [hunk 1]
```diff
@@ -41,6 +57,157 @@ class OwnedPathConflict:
...
+            status=workspace.status,
+            repo_url=workspace.repo_url,
+            base_branch=workspace.branch_base,
+            title=workspace.task_title,
+            task_class=workspace.task_class,
+            owned_paths=list(workspace.owned_paths),
+        )
+        self._session.add(attempt)
+        await self._session.flush()
+        return attempt
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/db/repositories.py:164]: fixed in commit a7adda57152afe2596ff0509760681ce7db6f671
